src/cdomain/value/cdomains/floatDomain.ml

```diff
@@ -708,14 +705,14 @@ module FloatIntervalImpl(Float_t : CFloatType) = struct
     | (l, h) when l = h && l = Float_t.of_float Nearest 1. -> of_const 0. (*acos(1) = 0*)
     | (l, h) when l < (Float_t.of_float Down (-.1.)) || h > (Float_t.of_float Up 1.) ->
       Messages.warn ~category:Messages.Category.Float "Domain error might occur: acos argument might be outside of [-1., 1.]";
-      of_interval (0., (overapprox_pi))
+      Interval (Float_t.of_float Down 0., Float_t.succ Float_t.pi)
     | (l, h) -> norm @@ Interval (Float_t.acos Down h, Float_t.acos Up l) (* acos is monotonic decreasing in [-1, 1]*)
 
   let eval_asin = function
     | (l, h) when l = h && l = Float_t.zero -> of_const 0. (*asin(0) = 0*)
     | (l, h) when l < (Float_t.of_float Down (-.1.)) || h > (Float_t.of_float Up 1.) ->
       Messages.warn ~category:Messages.Category.Float "Domain error might occur: asin argument might be outside of [-1., 1.]";
-      div (of_interval ((-. overapprox_pi), overapprox_pi)) (of_const 2.)
+      div (Interval (Float_t.pred (Float_t.neg Float_t.pi), Float_t.succ Float_t.pi)) (of_const 2.)
```

Comment:
I think no rounding is necessary since the pi float from `math.h` should be exactly representable and negation is exact since IEEE 754 floats have a sign bit.